tgz.go

```diff
@@ -4,14 +4,18 @@ import (
 	"archive/tar"
 	"bytes"
 	"compress/gzip"
+	"fmt"
 	"io"
+	"math/rand"
 	"time"
 )
 
 type Archive struct {
```

Comment:
@philbrookes could you document these types? Archive, ArchiveWriter. It's not very clear what is the responsibility of each of those to me.

If we have an "ArchiveWriter", shouldn't it be the one responsible for serializing writes?!